cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/EnforcementResult.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+public interface EnforcementResult {
```

Comment:
Can we name it `AuthorizationResult` and enum `AuthorizationStatus {AUTHORIZED,UNAUTHORIZED,NOT_REQUIRED}`? The enforcement does not happen here, only authorization

Reply:
I am also wondering if this should be a POJO class with a builder? Otherwise any extension would need to implement it. Builder help with changes retaining backward compatibilty

Reply:
You also probably want to include everything from UnauthorizedException, so that you could reconstruct it